src/Microsoft.ML.Fairlearn/metrics/FairlearnMetricCatalog.cs

```diff
@@ -0,0 +1,269 @@
+            result["AreaUnderPrecisionRecallCurve"] = DataFrameColumn.Create("AreaUnderPrecisionRecallCurve", groupMetric.Keys.Select(k => groupMetric[k].AreaUnderPrecisionRecallCurve));
+
+            return result;
+        }
+
+        public Dictionary<string, double> Overall()
+        {
+            CalibratedBinaryClassificationMetrics metrics = _context.BinaryClassification.Evaluate(_eval, _labelColumn);
+
+            // create the dictionary to hold the results
+            Dictionary<string, double> metricsDict = new Dictionary<string, double>
+            {
+                { "AUC", metrics.AreaUnderRocCurve },
+                { "Accuracy", metrics.Accuracy },
+                { "PosPrec", metrics.PositivePrecision },
+                { "PosRecall", metrics.PositiveRecall },
+                { "NegPrec", metrics.NegativePrecision },
+                { "NegRecall", metrics.NegativeRecall },
+                { "F1Score", metrics.F1Score },
+                { "AreaUnderPrecisionRecallCurve", metrics.AreaUnderPrecisionRecallCurve },
+                // following metrics are from the extensions
+                { "LogLoss", metrics.LogLoss },
+                { "LogLossReduction", metrics.LogLossReduction },
+                { "Entropy", metrics.Entropy }
+            };
+
+            return metricsDict;
+        }
+
+        private class ModelInput
+        {
+            public bool Label { get; set; }
+
+            public bool PredictedLabel { get; set; }
+
+            public float Score { get; set; }
+        }
+    }
+    public class RegressionGroupMetric : IGroupMetric
+    {
+        private readonly IDataView _eval;
+        private readonly string _labelColumn;
+        private readonly string _scoreColumn;
+        private readonly string _sensitiveFeatureColumn;
+        private readonly MLContext _context = new MLContext();
+
+        public RegressionGroupMetric(IDataView eval, string labelColumn, string scoreColumn, string sensitiveFeatureColumn)
+        {
+            _eval = eval;
+            _labelColumn = labelColumn;
+            _scoreColumn = scoreColumn;
+            _sensitiveFeatureColumn = sensitiveFeatureColumn;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public DataFrame ByGroup()
+        {
+            // 1. group row according to sensitive feature column
+            // 2. split dataset to different groups, data_g1, data_g2.....
+            // 3. calculate binary metrics for different groups
+            // 4. create datafrome from result of step 3
+            // 5. return it.
+            var sensitiveCol = _eval.Schema[_sensitiveFeatureColumn];
+            // get all the columns of the schema
+            DataViewSchema columns = _eval.Schema;
+
+            // TODO: is converting IDataview to DataFrame the best practice?
```

Comment:
Its not the best practice, especially when input files (depending on the trainer) can be streamed in and can be larger than the memory on the machine. Its probably fine to start out this way, but it will be something that is needed to change later.